tools/db/build/sp_keyboard_search.sql

```diff
@@ -42,7 +42,7 @@ AS
     name as name,
     case
       when name = @name or name_kd = @name then @weight_factor_exact_match -- exact match gets 3x weight factor
-      else 1 -- otherwise same weight
+      else 3-log10(len(name)) -- otherwise give slightly greater weight to shorter matches
```

Comment:
:+1:

We don't exactly have language names 100 chars long, let alone 1000, so this ought be fine.